File: Data Structures & Algorithms/house-robber/submission-0.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        
        int len = nums.size();
        vector<int> dp(nums.size() + 1);

        dp[0] = nums[0];
        for(int i=1;i<len;i++)
        {
            dp[i] = max(dp[i-2] + nums[i], dp[i-1]);
        }

        return dp[len-1];
    }
};

File: Data Structures & Algorithms/network-delay-time/submission-0.js
class Solution {
    /**
     * @param {number[][]} times
     * @param {number} n
     * @param {number} k
     * @return {number}
     */
    networkDelayTime(times, n, k) {
        
        let dist = Array(n+1).fill(Infinity);

        dist[k] = 0;

        for(let i=1; i<n; i++)
        {
            for(const [u,v,time] of times)
            {
                if(dist[u]!==Infinity && dist[u] + time < dist[v]) dist[v] = dist[u] + time;
            }
        }

        const res = Math.max(...dist.slice(1));

        return res == Infinity ? -1 : res;
    };
}
